src/settings/JobProfiles/JobProfileDetailsRoute/JobProfileDetailsRoute.js

```diff
@@ -1,32 +1,40 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 import {
   get,
   find,
 } from 'lodash';
 
 import { JobProfileDetails } from '../JobProfileDetails';
+import { DEFAULT_JOB_PROFILE_ID } from '../../../utils';
 
 const JobProfileDetailsRoute = ({
   resources: {
     jobProfile,
     mappingProfile,
+    jobExecutions,
   },
   history,
   location,
   match,
 }) => {
-  // `find` is used to make sure the matched job profile and mapping profile are displayed to avoid
+  // `find` is used to make sure the matched job profile, mapping profile and job executions are displayed to avoid
   // the flickering because of the disappearing of the previous and appearing of the new ones
   const jobProfileRecord = find([get(jobProfile, 'records.0', {})], { id: match.params.id });
   const mappingProfileRecord = find([get(mappingProfile, 'records.0', {})], { id: jobProfileRecord?.mappingProfileId });
+  const isDefaultProfile = jobProfileRecord?.id === DEFAULT_JOB_PROFILE_ID;
+  const isProfileAlreadyInUse = isDefaultProfile || Boolean(find([get(jobExecutions, 'records.0', {})], { jobProfileId: match.params.id }));
```

Comment:
I think isDefaultProfile doesn't mean that profile is in use. Maybe, it tells us that it can't be deleted. What do you think about it?

Reply:
I wanted to reduce the number of props that prevent the edit or deletion of the profile now and maybe something else in the future for the same reason. Initially, I had two `isDefaultProfile` and `isProfileAlreadyInUse` passed down to the JobProfileDeatils and hence to `JobProfileActionMenu`. So I ended up with only `isProfileAlreadyInUse` as default profile is kind of in use by being the default one. If you think of some better wording for that, please suggest.

Reply:
I prefer to use the clear naming instead of adding some props. Anyway, we can try to avoid using isProfileUsed and move to the buttons state. For example, isDeletedButtonEnabled or something similar.

Reply:
But the overall idea still won't be unresolved, in this case, the condition to disable the edit or delete button is based upon two factors: whether the profile is default or whether it is in use. In this case, both buttons would receive the kind of expression for the disabled state: `disabled: !isDefaultProfiel || !isProfileUsed`. My idea is to reduce the number of such conditions by coming with the props: `isProvfileAreadyInUse` or `isProfileUsed`. I don't think that introducing of the `isDeletedButtonEnabled` is worth doing as in future we would need to also provide additional props for buttons such callback what to do upon click, so it would require some additional thinking how to provide props for buttons if we up to reuse the ActionMenu component both for job profiles and mapping profiles.

Reply:
Please, could you clarify how isProvfileAreadyInUse reduces amount of conditions? It is used in on place. 
`disabled: !isDefaultProfile || !isProfileUsed` is used in one place and more clear for understanding

Reply:
Two conditions for each button. I am okay to introduce this approach if current is not readable enough for you. I was thinking that the `ifProfileAlreadyInUse` is "kind" of that one but okay with two props approach as well.

Reply:
Okay, in this case. Please, provide more clear name

Reply:
I already working on two props approach actually, since you pointed out to this initially I think it makes sense doing it from a readability perspective. Moreover, I don't know what a more readable name would be.

Reply:
This is not a change request and you can use 1 field. But you should clarify the default state in the variable name.